Exploring data. What is in there?

Controlliamo il CSV:

In [2]:
import csv
dirPath = '/Users/albertocottica/Documents/GitHub/-milano-melting-pot/dati/'
data = [] # we know from JSON that 
with open(dirPath + 'ds27_pop_sto_quartiere.csv') as datafile:
    reader = csv.DictReader(datafile, delimiter=';')
    for row in reader:
        drow = dict(row)
        drow['Anno'] = drow.pop('\ufeffAnno') # questo è per togliere spazzatura che confonde il nome del campo
        data.append(drow)
print (len(data))

2136906


In [3]:
print(data[0])

{'IdNil': '1', 'Nil': 'Duomo', 'Eta': '0', 'Genere': 'Femmine', 'Cittadinanza': 'Cinese, Rep. Popolare', 'Residenti': '1', 'Anno': '1999'}


In [4]:
quartieri = [] # quanti e quali quartieri abbiamo?
missingRecords = 0 
for item in data:
    if 'Nil' in item:        
        if item['Nil'] not  in quartieri:
            quartieri.append(item['Nil'])
    else:
        missingRecords += 1

print (quartieri)
print ('Gli ID di quartiere a Milano sono ' + str(len(quartieri)))
print ('Bad records: ' + str(missingRecords) + ' su ' + str(len(data)))

['Duomo', 'Brera', 'Giardini Porta Venezia', 'Guastalla', 'Vigentina', 'Ticinese', 'Magenta - S. Vittore', 'Parco Sempione', 'Garibaldi Repubblica', 'Centrale', 'Isola', 'Maciachini - Maggiolina', 'Greco', 'Niguarda - Cà Granda', 'Bicocca', 'Vle Monza', 'Adriano', 'Parco Lambro - Cimiano', 'Padova', 'Loreto', 'Buenos Aires - Venezia', 'Città Studi', 'Lambrate', 'Parco Forlanini - Ortica', 'Corsica', 'XXII Marzo', 'Porta Romana', 'Umbria - Molise', 'Ortomercato', 'Mecenate', 'Parco Monlué - Ponte Lambro', 'Triulzo Superiore', 'Rogoredo', 'Chiaravalle', 'Lodi - Corvetto', 'Scalo Romana', 'Ex OM - Morivione', 'Ripamonti', 'Quintosole', 'Ronchetto delle Rane', 'Gratosoglio - Ticinello', 'Stadera', 'Tibaldi', 'Navigli', 'S. Cristoforo', 'Barona', 'Cantalupa', 'Ronchetto S/N', 'Giambellino', 'Tortona', 'Washington', 'Bande Nere', 'Lorenteggio', 'Muggiano', 'Baggio', 'Forze Armate', 'Selinunte', 'De Angeli - Monte Rosa', 'Tre Torri', 'S. Siro', 'Quarto Cagnino', 'Quinto Romano', 'Figino', 'Tr

In [5]:
anni = [] # quanti e quali anni abbiamo?
missingRecords = 0 
for item in data:
    if 'Anno' in item:        
        if item['Anno'] not  in anni:
            anni.append(item['Anno'])
    else:
        missingRecords += 1

print (anni)
print ('Anni nel dataset: ' + str(len(anni)))
print ('Bad records: ' + str(missingRecords) + ' su ' + str(len(data)))

['1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016', '2017', '2018']
Anni nel dataset: 20
Bad records: 0 su 2136906


Controlliamo che la popolazione corrisponda

In [6]:
pop = 0
nonItaliani = 0
for item in data:
    if item['Anno'] == '2018':
        pop += int(item['Residenti'])
        if item['Cittadinanza'] != 'Italia':
            nonItaliani += int(item['Residenti'])
print(pop)
print (nonItaliani)

1395274
275818


In [7]:
nazioni = [] # quante e quali nazionalità abbiamo?
missingRecords = 0 
for item in data:
    if 'Cittadinanza' in item:        
        if item['Cittadinanza'] not  in nazioni:
            nazioni.append(item['Cittadinanza'])
    else:
        missingRecords += 1

print (nazioni)
print ('Le nazionalità presenti a Milano sono ' + str(len(nazioni)))
print ('Bad records: ' + str(missingRecords) + ' su ' + str(len(data)))

['Cinese, Rep. Popolare', 'Egitto', 'Etiopia', 'Filippine', 'Italia', 'Sri Lanka (ex Ceylon)', 'Togo', 'Perù', 'Francia', 'Libano', 'Mauritius', 'Regno Unito', 'India', 'Spagna', "Stati Uniti D'America", 'Israele', 'Turchia', 'Grecia', 'Marocco', 'Portogallo', 'Germania', 'Giappone', 'El Salvador', 'Iran', 'Bulgaria', 'Ecuador', 'Croazia', 'Bosnia-Erzegovina', 'ex Jugoslavia', 'Albania', 'Siria', 'Argentina', 'Cuba', 'Georgia', 'Dominicana, Rep.', 'Svizzera', 'Colombia', 'Eritrea', 'Romania', 'Russa, Federazione', 'Somalia', 'Ucraina', 'Venezuela', 'Belgio', 'Danimarca', 'Irlanda', 'Ungheria', 'Brasile', 'Svezia', 'Australia', 'Austria', 'Estonia', 'Dominica', 'Taiwan (ex Formosa)', 'Tunisia', 'Capo Verde', 'Ghana', 'Kenya', 'Lituania', 'Nigeria', 'Norvegia', 'Paesi Bassi', 'Ruanda', 'Thailandia', 'Giordania', 'Finlandia', 'Cile', 'Indonesia', 'Canada', 'Haiti', 'Nepal', 'Pakistan', 'Messico', 'Polonia', 'Corea del Sud, Rep.', 'Bolivia', 'Uruguay', 'Costa Rica', 'Senegal', 'Lussemburgo

A questo punto possiamo pre-aggregare i dati. 

Ci interessa quante persone abbiamo di ciascuna nazionalità, in ciascun quartiere e in ciascun anno. Qundi dobbiamo sommare la variabile `'Residenti'` per ciascuna tripla di valori sulle variabili `'Anno'`, `'Nil'` e `'Cittadinanza'`. Inoltre, non dobbiamo perderci il genere.

Mi faccio un dizionario di liste di dizionari.

In [7]:
nestedData = {}
for quartiere in quartieri:
    nestedData[quartiere] = []
    for item in data:
        if item['Nil'] == quartiere:
            nestedData[quartiere].append(item)

In [8]:
test = nestedData['Duomo']

print (len(test))

31858


In [28]:
import datetime
start_script = datetime.datetime.now()
summedData = []
for key in nestedData:
    y = nestedData[key]
    print('++++++ '+ str(key))
    for anno in anni:
        print (anno)
        for nazione in nazioni: 
            triple = {'Anno': anno, 'Nil': key, 'Cittadinanza': nazione, 'Maschi': 0, 'Femmine': 0}
            for item in y:
                now = item ['Anno']
                people = item  ['Cittadinanza']
                if anno == now and people == nazione:
                    if item['Genere'] == 'Femmine':
                        triple['Femmine'] += int(item ['Residenti'])
                    else: 
                        triple['Maschi'] += int(item['Residenti'])
            summedData.append(triple)

print (summedData[0])
print (len(summedData))

end_script = datetime.datetime.now()
print ('Completato in ' + str(end_script-start_script))

++++++ Duomo
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Brera
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Giardini Porta Venezia
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Guastalla
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Vigentina
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Ticinese
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Magenta - S. Vittore
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Parco Sempione
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Garibaldi Repubblica
1999
2000
2001
2

2011
2012
2013
2014
2015
2016
2017
2018
++++++ Ghisolfa
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Villapizzone
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Maggiore - Musocco
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Cascina Triulza - Expo
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Sacco
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Stephenson
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Quarto Oggiaro
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
++++++ Bovisa
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
+

Testiamo per vedere se funziona bene:

In [29]:
for i in range(10):
    print(summedData[i])
print (len(summedData))

{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Cinese, Rep. Popolare', 'Maschi': 14, 'Femmine': 11}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Egitto', 'Maschi': 40, 'Femmine': 12}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Etiopia', 'Maschi': 9, 'Femmine': 22}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Filippine', 'Maschi': 97, 'Femmine': 205}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Italia', 'Maschi': 8653, 'Femmine': 8948}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Sri Lanka (ex Ceylon)', 'Maschi': 91, 'Femmine': 76}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Togo', 'Maschi': 0, 'Femmine': 2}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Perù', 'Maschi': 21, 'Femmine': 57}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Francia', 'Maschi': 100, 'Femmine': 91}
{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Libano', 'Maschi': 21, 'Femmine': 10}
309760


Voglio aggregare in macro categorie. In prima approssimazione, manteniamo la classificazione del progetto [Brussels Melting Pot](http://brussels-diversity.jetpack.ai/), e usiamo i nomi inglesi delle categorie per mantenere la comparabilità con esso. Questa prevede due livelli di aggregazione: regioni e continenti. Per prima cosa costruisco liste dei paesi che appartengono a ciascuna regione. Le regioni sono:

* EU 15 ([definizione](https://stats.oecd.org/glossary/detail.asp?ID=6805)), esclusa l'Italia
* EU N13, cioè i 28 paesi dell'UE meno i 15 di EU 15 
* Non-EU Europe (questo non c'è nel progetto Bruxelles)
* North Africa 
* Sub-Saharian Africa
* Eastern Asia ([definizione](https://en.wikipedia.org/wiki/East_Asia))
* Southern Asia
* Southeast Asia (non c'è nel progetto di Bruxelles, se i numeri sono piccoli lo elimino dopo)
* Western Asia
* North America
* South America
* Oceania
* Other (residuo)

I continenti sono:

* Europe
* Asia
* Africa
* North America
* South America
* Oceania

(il residuo è omesso)

In [40]:
eu15 = ['Francia', 'Regno Unito', 'Spagna', 'Grecia', 'Portogallo', 'Germania', 'Belgio', 'Danimarca', 'Irlanda', 
        'Svezia', 'Austria', 'Paesi Bassi', 'Finlandia', 'Lussemburgo', 'Andorra', 'Monaco', 'Liechtenstein']
eun13 = ['Bulgaria', 'Croazia', 'Romania', 'Ungheria', 'Estonia', 'Lituania', 'Polonia', 'Slovenia', 'Ceca, Rep.', 
         'Slovacchia', 'Lettonia', 'Malta', 'Cipro']
otheu = ['Bosnia-Erzegovina', 'ex Jugoslavia', 'Albania', 'Georgia', 'Svizzera', 'Russa, Federazione', 'Ucraina', 
         'Norvegia', 'Moldova', 'Armenia', 'Islanda', 'Bielorussia', 'Azerbaigian', 'Kosovo', 'Serbia', 'Montenegro', 
         'Macedonia, ex Rep. Jugoslava']
noraf = ['Egitto', 'Marocco', 'Tunisia', 'Libia', 'Algeria']
ssaf = ['Etiopia', 'Togo', 'Mauritius', 'Eritrea', 'Somalia', 'Capo Verde', 'Ghana', 'Kenya', 'Nigeria', 'Ruanda', 
        'Senegal', 'Tanzania', 'Seychelles', 'Sudan', 'Sud Africa', "Costa D'Avorio", 'Camerun', 
        'Benin (ex Dahomey)', 'Burkina Faso (ex Alto Volta)', 'Zambia', 'Angola', 'Guinea', 'Gambia', 'Congo', 
        'Sierra Leone', 'Uganda', 'Burundi', 'Madagascar', 'Congo Rep. Dem (ex Zaire)', 'Gabon', 'Liberia', 
        'Mozambico', 'Mauritania', 'Mali', 'Zimbabwe (ex Rhodesia)', 'Gibuti', 'Niger', 'Ciad', 'Guinea Bissau', 
        'Centrafricana, Rep.', 'Sao Tomè e Principe', 'Malawi', 'Swaziland', 'Lesotho', 'Guinea Equatoriale', 
         'Sud Sudan, Repubblica del']
eastas = ['Cinese, Rep. Popolare', 'Giappone', 'Taiwan (ex Formosa)', 'Corea del Sud, Rep.', 
          'Corea del Nord, Rep. Pop. Dem', 'Mongolia']
southas = ['India', 'Bangladesh', 'Afghanistan', 'Bhutan', 'Maldive', 'Pakistan', 'Sri Lanka (ex Ceylon)', 'Nepal']
seas = ['Indonesia', 'Malaysia', 'Vietnam', 'Singapore', 'Myanmar (ex Birmania)', 'Laos', 'Cambogia', 
        'Timor Orientale', 'Thailandia', 'Filippine']
westas = ['Libano', 'Israele', 'Turchia', 'Iran', 'Siria', 'Iraq', 'Kazakistan', 'Uzbekistan', 'Yemen', 
          'Territori Autonomia Palestinese', 'Kuwait', 'Turkmenistan', 'Tagikistan', 'Kirghizistan', 'Oman']
noram = ["Stati Uniti D'America", 'Canada', 'Giordania', 'Arabia Saudita']
soam = ['Messico', 'Perù', 'El Salvador', 'Ecuador', 'Argentina', 'Cuba', 'Dominicana, Rep.', 'Colombia', 
        'Venezuela', 'Brasile', 'Dominica', 'Cile', 'Haiti', 'Bolivia', 'Uruguay', 'Costa Rica', 'Giamaica', 
       'Guatemala', 'Panama', 'Trinidad e Tobago', 'Nicaragua', 'Antigua e Barbuda', 'Suriname', 'Honduras', 
       'Bahamas', 'Barbados', 'Belize', 'Saint Kitts e Nevis', 'Paraguay']
ocean = ['Australia', 'Nuova Zelanda']
other = ['Altre nazioni o n.d.', 'Apolide']


check = len(eu15) + len(eun13) + len(otheu) + len(noraf) + len(ssaf) + len(eastas) + len(southas) + len(seas) \
        + len(westas) + len(noram) + len(soam) + len(ocean) + len(other)

def missing(list1, list2):
    '''
    returns a list of items that are included in list1, but not in list 2
    '''
    missing_stuff = []
    for item in list1:
        if item not in list2:
            missing_stuff.append(item)
    return missing_stuff

print(check + 1)
print(len(nazioni))
check = missing(nazioni, eu15 + eun13 + otheu + noraf + ssaf + eastas + southas + seas + westas + noram + soam \
               + ocean + other)
print(check)


175
176
['Italia', 'San Marino']


Rimangono fuori solo Italia e San Marino, che in effetti contano come indigeni a Milano. 

Aggiungo regioni e continenti al dataset. 

In [51]:
def assign_country(country):
    '''
    returns a tuple with the region and continent of country
    '''
    if country in eu15:
        location = ('eu15', 'europa')
    elif country in eun13:
        location = ('eun13', 'europa')
    elif country in otheu:
        location = ('otheu', 'europa')
    elif country in noraf:
        location = ('noraf', 'africa')
    elif country in ssaf:
        location = ('ssaf', 'africa')
    elif country in eastas:
        location = ('eastas', 'asia')
    elif country in southas:
        location = ('southas', 'asia')
    elif country in seas:
        location = ('seas', 'asia')
    elif country in westas:
        location = ('westas', 'asia')
    elif country in noram:
        location = ('noram', 'north america')
    elif country in soam:
        location = ('soam', 'south america')
    elif country in ocean:
        location = ('ocean', 'oceania')
    elif country in other:
        location = ('other','')
    else: 
        location = ('ita', 'italia')
    return location

for item in summedData:
    nazione = item['Cittadinanza']
    item['regione'] = assign_country(nazione)[0]
    item['continente'] = assign_country(nazione)[1]

print (summedData[:4]) ## just for testing  

[{'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Cinese, Rep. Popolare', 'Maschi': 14, 'Femmine': 11, 'regione': 'eastas', 'continente': 'asia'}, {'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Egitto', 'Maschi': 40, 'Femmine': 12, 'regione': 'noraf', 'continente': 'africa'}, {'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Etiopia', 'Maschi': 9, 'Femmine': 22, 'regione': 'ssaf', 'continente': 'africa'}, {'Anno': '1999', 'Nil': 'Duomo', 'Cittadinanza': 'Filippine', 'Maschi': 97, 'Femmine': 205, 'regione': 'seas', 'continente': 'asia'}]


Infine, salviamo in CSV.

In [52]:
import csv
with open (dirPath + 'summedData.csv', 'w') as outfile:
    fieldnames = ['Anno', 'Nil', 'Cittadinanza', 'regione', 'continente', 'Maschi', 'Femmine']
    writer = csv.DictWriter(outfile, fieldnames=fieldnames)
    writer.writeheader()
    for item in summedData:
        writer.writerow(item)
print('Done')

Done


Per comodità, mi scrivo anche un dataset "compatto", con un record per regione, quartiere e anno. In questo modo le 176 nazionalità vengono compattate in 15 regioni.

In [63]:
compactData = []
for anno in anni:
    print (anno)
    for quartiere in quartieri:
        print (quartiere)
        newrecord ={'Anno': anno, 'Nil': quartiere}
        for item in summedData: 
            if item['Anno'] == anno and item['Nil'] == quartiere:
                newkey = item['regione'] + ' M'
                if newkey not in newrecord:
                    newrecord[newkey] = item['Maschi']
                    newrecord[item['regione'] + ' F'] = item['Femmine']
                else:
                    newrecord[item['regione'] + ' M'] += item['Maschi']
                    newrecord[item['regione'] + ' F'] += item['Femmine']
        compactData.append(newrecord)

print(compactData[:4])

1999
Duomo
Brera
Giardini Porta Venezia
Guastalla
Vigentina
Ticinese
Magenta - S. Vittore
Parco Sempione
Garibaldi Repubblica
Centrale
Isola
Maciachini - Maggiolina
Greco
Niguarda - Cà Granda
Bicocca
Vle Monza
Adriano
Parco Lambro - Cimiano
Padova
Loreto
Buenos Aires - Venezia
Città Studi
Lambrate
Parco Forlanini - Ortica
Corsica
XXII Marzo
Porta Romana
Umbria - Molise
Ortomercato
Mecenate
Parco Monlué - Ponte Lambro
Triulzo Superiore
Rogoredo
Chiaravalle
Lodi - Corvetto
Scalo Romana
Ex OM - Morivione
Ripamonti
Quintosole
Ronchetto delle Rane
Gratosoglio - Ticinello
Stadera
Tibaldi
Navigli
S. Cristoforo
Barona
Cantalupa
Ronchetto S/N
Giambellino
Tortona
Washington
Bande Nere
Lorenteggio
Muggiano
Baggio
Forze Armate
Selinunte
De Angeli - Monte Rosa
Tre Torri
S. Siro
Quarto Cagnino
Quinto Romano
Figino
Trenno
Gallaratese
QT 8
Portello
Pagano
Sarpi
Ghisolfa
Villapizzone
Maggiore - Musocco
Cascina Triulza - Expo
Sacco
Stephenson
Quarto Oggiaro
Bovisa
Farini
Dergano
Affori
Bovisasca
Comasin

Navigli
S. Cristoforo
Barona
Cantalupa
Ronchetto S/N
Giambellino
Tortona
Washington
Bande Nere
Lorenteggio
Muggiano
Baggio
Forze Armate
Selinunte
De Angeli - Monte Rosa
Tre Torri
S. Siro
Quarto Cagnino
Quinto Romano
Figino
Trenno
Gallaratese
QT 8
Portello
Pagano
Sarpi
Ghisolfa
Villapizzone
Maggiore - Musocco
Cascina Triulza - Expo
Sacco
Stephenson
Quarto Oggiaro
Bovisa
Farini
Dergano
Affori
Bovisasca
Comasina
Bruzzano
Parco Nord
Parco delle Abbazie
Parco dei Navigli
Parco Agricolo Sud
Parco Bosco in Città
2007
Duomo
Brera
Giardini Porta Venezia
Guastalla
Vigentina
Ticinese
Magenta - S. Vittore
Parco Sempione
Garibaldi Repubblica
Centrale
Isola
Maciachini - Maggiolina
Greco
Niguarda - Cà Granda
Bicocca
Vle Monza
Adriano
Parco Lambro - Cimiano
Padova
Loreto
Buenos Aires - Venezia
Città Studi
Lambrate
Parco Forlanini - Ortica
Corsica
XXII Marzo
Porta Romana
Umbria - Molise
Ortomercato
Mecenate
Parco Monlué - Ponte Lambro
Triulzo Superiore
Rogoredo
Chiaravalle
Lodi - Corvetto
Scalo Romana


Ticinese
Magenta - S. Vittore
Parco Sempione
Garibaldi Repubblica
Centrale
Isola
Maciachini - Maggiolina
Greco
Niguarda - Cà Granda
Bicocca
Vle Monza
Adriano
Parco Lambro - Cimiano
Padova
Loreto
Buenos Aires - Venezia
Città Studi
Lambrate
Parco Forlanini - Ortica
Corsica
XXII Marzo
Porta Romana
Umbria - Molise
Ortomercato
Mecenate
Parco Monlué - Ponte Lambro
Triulzo Superiore
Rogoredo
Chiaravalle
Lodi - Corvetto
Scalo Romana
Ex OM - Morivione
Ripamonti
Quintosole
Ronchetto delle Rane
Gratosoglio - Ticinello
Stadera
Tibaldi
Navigli
S. Cristoforo
Barona
Cantalupa
Ronchetto S/N
Giambellino
Tortona
Washington
Bande Nere
Lorenteggio
Muggiano
Baggio
Forze Armate
Selinunte
De Angeli - Monte Rosa
Tre Torri
S. Siro
Quarto Cagnino
Quinto Romano
Figino
Trenno
Gallaratese
QT 8
Portello
Pagano
Sarpi
Ghisolfa
Villapizzone
Maggiore - Musocco
Cascina Triulza - Expo
Sacco
Stephenson
Quarto Oggiaro
Bovisa
Farini
Dergano
Affori
Bovisasca
Comasina
Bruzzano
Parco Nord
Parco delle Abbazie
Parco dei Navigli


Salvo anche questo:

In [64]:
with open (dirPath + 'compactData.csv', 'w') as outfile:
    fieldnames = ['Anno', 'Nil', 'eastas M', 'eastas F', 'noraf M', 'noraf F', 'ssaf M', 
                  'ssaf F', 'seas M', 'seas F', 'ita M', 'ita F', 'southas M', 
                  'southas F', 'soam M', 'soam F', 'eu15 M', 'eu15 F', 'westas M', 
                  'westas F', 'noram M', 'noram F', 'eun13 M', 'eun13 F', 
                  'otheu M', 'otheu F', 'ocean M', 'ocean F', 'other M', 'other F']
    writer = csv.DictWriter(outfile, fieldnames=fieldnames)
    writer.writeheader()
    for item in compactData:
        writer.writerow(item)
print('Done')

Done


Proviamo a identificare i 